localstripe/resources.py

```diff
@@ -1400,6 +1395,70 @@ def _api_list_all(cls, url, customer=None, limit=None):
         return li
 
 
+class InvoiceLineItem(StripeObject):
+    object = 'line_item'
+    _id_prefix = 'il_'
+
+    def __init__(self, item):
+        try:
+            assert isinstance(item, (InvoiceItem, SubscriptionItem))
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        # All exceptions must be raised before this point.
+        super().__init__()
+
+        self.type = \
+            'invoiceitem' if isinstance(item, InvoiceItem) else 'subscription'
+
+        if self.type == 'subscription':
+            self.subscription_item = item.id
+            self.subscription = item._subscription
+        elif self.type == 'invoiceitem':
+            self.invoice_item = item.id
+            self.subscription = None
+
+        # Legacy support, before InvoiceLineItem
+        self.invoice = item.invoice
+
+        self.plan = item.plan
+        self.tax_rates = item.tax_rates or []
+        self.metadata = item.metadata
+
+        self.amount = item.amount \
+            if self.type == 'invoiceitem' else item._calculate_amount()
+        self.currency = \
+            item.currency if self.type == 'invoiceitem' else item.plan.currency
+        self.description = \
+            item.description if self.type == 'invoiceitem' else item.plan.name
+        self.proration = \
+            item.proration if self.type == 'invoiceitem' else False
+
+        self.quantity = item.quantity if self.type == 'subscription' else 1
```

Comment:
In `localstripe` the quantity can only be set on `SubscriptionItem`s so I use `1` as the default value for `InvoiceItem`s.

Reply:
Perfect! It seems in agreement with what Stripe API doc examples show.

Reply:
Please note that `InvoiceItem` has a `quantity` field (see the [doc](https://stripe.com/docs/api/invoiceitems/object#invoiceitem_object-quantity)) but localstripe do not implement it. Maybe a comment in the code can help better understand this default value?

Reply:
Ah, thanks, I hadn't seen that!

Maybe a semi-quick-win would be to add this in `InvoiceItem`:
```diff
@@ -1368,6 +1368,7 @@ class InvoiceItem(StripeObject):
         self.invoice = invoice
         self.subscription = subscription
         self.plan = plan
+        self.quantity = 1
         self.amount = amount
         self.currency = currency
         self.customer = customer
```
and just use this in `InvoiceLineItem`:
```python
self.quantity = item.quantity
```
?

Reply:
Good idea. It would be clearer on why this number is 1 by default. Code updated.